cooked-validators/src/Cooked/MockChain/Monad.hs

```diff
@@ -86,15 +86,15 @@ class (MonadFail m) => MonadBlockChain m where
   awaitTime :: Pl.POSIXTime -> m Pl.POSIXTime
 
 -- | Calls 'validateTxSkel' with a skeleton that is set with some specific options.
-validateTxConstrOpts :: (MonadBlockChain m) => TxOpts -> [Constraint] -> m Pl.TxId
```

Comment:
It would be really nice if we don't have to change the types of these functions. I think we should still be able to construct a transaction from a list of constraint; All we need is to have a function: `unorderedTxSpecFromConstraint :: [Constraint] -> TxSpec`.

Reply:
I also think this would make it much easier to adapt the tests.